app/src/main/java/com/google/samples/apps/sunflower/adapters/SunflowerPagerAdapter.kt

```diff
@@ -17,31 +17,26 @@
 package com.google.samples.apps.sunflower.adapters
 
 import androidx.fragment.app.Fragment
-import androidx.fragment.app.FragmentManager
-import androidx.fragment.app.FragmentStatePagerAdapter
+import androidx.viewpager2.adapter.FragmentStateAdapter
 import com.google.samples.apps.sunflower.GardenFragment
 import com.google.samples.apps.sunflower.PlantListFragment
-import kotlin.IndexOutOfBoundsException
 
 const val MY_GARDEN_PAGE_INDEX = 0
 const val PLANT_LIST_PAGE_INDEX = 1
 
-class SunflowerPagerAdapter(
-    private val tabTitles: HashMap<Int, String>,
-    private val tabFragments: HashMap<Int, Fragment>,
-    fragmentManager: FragmentManager
-) : FragmentStatePagerAdapter(fragmentManager, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
+class SunflowerPagerAdapter(fragment: Fragment) : FragmentStateAdapter(fragment) {
 
     /**
-     * There are only two fragments in this ViewPager: [GardenFragment] and [PlantListFragment]
+     * Mapping of the ViewPager page indexes to their respective Fragments
      */
-    override fun getCount() = 2
+    private val tabFragments = hashMapOf(
+        MY_GARDEN_PAGE_INDEX to GardenFragment(),
+        PLANT_LIST_PAGE_INDEX to PlantListFragment()
+    )
 
-    override fun getItem(position: Int): Fragment {
-        return tabFragments[position] ?: throw IndexOutOfBoundsException()
-    }
+    override fun getItemCount() = tabFragments.size
 
-    override fun getPageTitle(position: Int): CharSequence? {
-        return tabTitles[position]
+    override fun createFragment(position: Int): Fragment {
+        return tabFragments[position] ?: throw IndexOutOfBoundsException()
```

Comment:
This might work, but the idea behind createFragment is that you return a new Fragment (i.e. create, not get). We try to stress that in JavaDoc.

My concern is that by leaving as is someone might follow this pattern and fall into subtle bugs.

Easy fix: replace tabFragments with tabFragmentCreators:
```
    val tabFragmentCreators : Map<Int, () -> Fragment> = hashMapOf(
        MY_GARDEN_PAGE_INDEX to {GardenFragment()},
        PLANT_LIST_PAGE_INDEX to {PlantListFragment()}
    )
```

Then: ` return tabFragments[position] -> return tabFragments[position].?invoke()`

Reply:
actually, I prefer `SpareseArray` or its compat version

Reply:
As long as `createFragment` creates new instances when called I'm happy.
For simplicity, `hashMapOf` -> `mapOf` should be fine IMO, but it's your call.

Reply:
Agreed, `mapOf` is a bit better than `SparseArray` for readability.

Updated to create a new Fragment.  Thanks @gielzak !